src/mlpack/methods/ann/layer/mean_pooling_impl.hpp

```diff
@@ -31,12 +31,19 @@ MeanPooling<InputDataType, OutputDataType>::MeanPooling(
     const size_t kernelHeight,
     const size_t strideWidth,
     const size_t strideHeight,
-    const bool floor) :
+    const bool floor,
+    const size_t padW,
+    const size_t padH,
```

Comment:
You can remove duplicated code. Have a look at [this](https://github.com/mlpack/mlpack/pull/2163/files#diff-5084090d262ede972b322a972db5664bR67).

Reply:
Hm, that's what I was trying to address [here](https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/ann_layer_test.cpp#L2903). Alternative could be delegate constructor with added overhead of tuple creation.

Reply:
Exactly.